File: apps/mobile-flutter/lib/features/obat/models/obat.dart
class Obat {
  const Obat({
    required this.id,
    required this.name,
    required this.type,
    required this.dosage,
    required this.usageInfo,
    required this.frequency,
    this.createdAt,
    this.notes,
    this.sideEffects,
    this.userId,
  });

  final String id;
  final String? userId;
  final String name;
  final String type;
  final String dosage;
  final String usageInfo;
  final String frequency;
  final String? sideEffects;
  final String? notes;
  final DateTime? createdAt;

  factory Obat.fromJson(Map<String, Object?> json) {
    return Obat(
      id: '${json['id'] ?? ''}',
      userId: json['user_id'] == null ? null : '${json['user_id']}',
      name: '${json['name'] ?? ''}',
      type: '${json['type'] ?? 'Lainnya'}',
      dosage: '${json['dosage'] ?? ''}',
      usageInfo: '${json['usage_info'] ?? ''}',
      frequency: '${json['frequency'] ?? ''}',
      sideEffects: json['side_effects'] == null ? null : '${json['side_effects']}',
      notes: json['notes'] == null ? null : '${json['notes']}',
      createdAt: DateTime.tryParse('${json['created_at'] ?? ''}'),
    );
  }

  Map<String, Object?> toJson() {
    return {
      'id': id,
      'user_id': userId,
      'name': name,
      'type': type,
      'dosage': dosage,
      'usage_info': usageInfo,
      'frequency': frequency,
      'side_effects': sideEffects,
      'notes': notes,
      'created_at': createdAt?.toIso8601String(),
    };
  }
}

File: packages/core/src/domain/obat.ts
export type ObatItem = {
  id: string;
  user_id?: string;
  name: string;
  type: string;
  dosage: string;
  usage_info: string;
  frequency: string;
  side_effects?: string | null;
  notes?: string | null;
  created_at?: string;
};

export type ObatInput = Pick<
  ObatItem,
  'name' | 'type' | 'dosage' | 'usage_info' | 'frequency' | 'side_effects' | 'notes'
>;

export function normalizeObatItem(input: Partial<ObatItem>): ObatItem {
  return {
    id: String(input.id ?? ''),
    user_id: input.user_id ? String(input.user_id) : undefined,
    name: String(input.name ?? ''),
    type: String(input.type ?? 'Lainnya'),
    dosage: String(input.dosage ?? ''),
    usage_info: String(input.usage_info ?? ''),
    frequency: String(input.frequency ?? ''),
    side_effects: input.side_effects == null ? null : String(input.side_effects),
    notes: input.notes == null ? null : String(input.notes),
    created_at: input.created_at ? String(input.created_at) : undefined,
  };
}

export function normalizeObatInput(input: Partial<ObatInput>): ObatInput {
  return {
    name: String(input.name ?? '').trim(),
    type: String(input.type ?? 'Lainnya').trim() || 'Lainnya',
    dosage: String(input.dosage ?? '').trim(),
    usage_info: String(input.usage_info ?? '').trim(),
    frequency: String(input.frequency ?? '').trim(),
    side_effects: input.side_effects == null ? null : String(input.side_effects).trim(),
    notes: input.notes == null ? null : String(input.notes).trim(),
  };
}
